object.go

```diff
@@ -0,0 +1,259 @@
+package httpapi
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"io/ioutil"
+
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-ipld-format"
+	"github.com/ipfs/go-merkledag"
+	"github.com/ipfs/interface-go-ipfs-core"
+	caopts "github.com/ipfs/interface-go-ipfs-core/options"
+)
+
+type ObjectAPI HttpApi
+
+type objectOut struct {
+	Hash string
+}
+
+func (api *ObjectAPI) New(ctx context.Context, opts ...caopts.ObjectNewOption) (format.Node, error) {
+	options, err := caopts.ObjectNewOptions(opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	var out objectOut
+	err = api.core().request("object/new", options.Type).Exec(ctx, &out)
+	if err != nil {
+		return nil, err
+	}
+
+	c, err := cid.Parse(out.Hash)
+	if err != nil {
+		return nil, err
+	}
+
+	return api.core().nodeFromPath(ctx, iface.IpfsPath(c)), nil
```

Comment:
Should this return the node or the CID? If we want to actually return a node, we could just return an entirely local node. Really, `object/new` doesn't _create_ anything. It just returns one of the template nodes.